test/e2e/common/machinepoolnamelease.go

```diff
@@ -0,0 +1,78 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"github.com/openshift/hive/pkg/constants"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/watch"
+	clientwatch "k8s.io/client-go/tools/watch"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+	"time"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+
+	log "github.com/sirupsen/logrus"
+
+	"k8s.io/client-go/rest"
+	clientcache "k8s.io/client-go/tools/cache"
+)
+
+// WaitForMachinePoolNameLease waits for a lease to appear in the given namespace for the given pool.
+func WaitForMachinePoolNameLease(config *rest.Config, namespace, poolName string, timeOut time.Duration) (string, error) {
+	logger := log.WithField("pool", poolName)
+	logger.Infof("Waiting for MachinePoolNameLease")
+	scheme, err := hivev1.SchemeBuilder.Build()
+
+	waitContext, cancel := context.WithTimeout(context.Background(), timeOut)
+	defer cancel()
+
+	gvk, err := apiutil.GVKForObject(&hivev1.MachinePoolNameLease{}, scheme)
```

Comment:
You do not need all the scheme and codec cruft.

```
hiveClient, err := hiveclient.NewForConfig(config)
if err != nil {
  return "", errors.Wrap(err, "could not create the Hive client")
}

restClient := hiveClient.HiveV1().RESTClient()
```

Reply:
Excellent, thanks.